x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/action/EqlSearchResponse.java

```diff
@@ -179,6 +182,110 @@ public String toString() {
         return Strings.toString(this);
     }
 
+    // Event
+    public static class Event implements Writeable, ToXContentObject {
+
+        private static final class Fields {
+            static final String INDEX = GetResult._INDEX;
+            static final String ID = GetResult._ID;
+            static final String SOURCE = SourceFieldMapper.NAME;
+        }
+
+        private static final ParseField INDEX = new ParseField(Fields.INDEX);
+        private static final ParseField ID = new ParseField(Fields.ID);
+        private static final ParseField SOURCE = new ParseField(Fields.SOURCE);
+
+        private static final ConstructingObjectParser<Event, Void> PARSER =
+                new ConstructingObjectParser<>("eql/search_response_event", true,
+                        args -> new Event((String) args[0], (String) args[1], (BytesReference) args[2]));
+
+        static {
+            PARSER.declareString(constructorArg(), INDEX);
+            PARSER.declareString(constructorArg(), ID);
+            PARSER.declareObject(constructorArg(), (p, c) -> {
+                try (XContentBuilder builder = XContentBuilder.builder(p.contentType().xContent())) {
+                    builder.copyCurrentStructure(p);
+                    return BytesReference.bytes(builder);
+                }
+            }, SOURCE);
+        }
+
+        private final String index;
+        private final String id;
+        private final BytesReference source;
+
+        public Event(String index, String id, BytesReference source) {
+            this.index = index;
+            this.id = id;
+            this.source = source;
+        }
+        
+        public Event(StreamInput in) throws IOException {
+            index = in.readString();
+            id = in.readString();
+            source = in.readBytesReference();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(index);
+            out.writeString(id);
+            out.writeBytesReference(source);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field(Fields.INDEX, index);
+            builder.field(Fields.ID, id);
+            //XContentHelper.writeRawField(Fields.SOURCE, source, builder.contentType(), builder, params);
+            XContentHelper.writeRawField(Fields.SOURCE, source, builder, params);
```

Comment:
I don't think we can avoid using this deprecated method as long as it is used by [SearchHit](https://github.com/elastic/elasticsearch/blob/1a4f89a69982384cb1023a496e5b8436b8139d8e/server/src/main/java/org/elasticsearch/search/SearchHit.java#L653) for the same basic reason it is used by SearchHit - we have no idea in which format the source was stored so we need to detect it in the run time. We would basically need to store original source content type in order to avoid using this method, which we don't.